### PR TITLE
 Feature :: Add jinja to render parameters in query and permissions

### PR DESCRIPTION
**Pull request by virobidou**

### CODE REVIEW

File: toucan_connectors/common.py [hunk 1]
```diff
@@ -10,25 +13,125 @@ def nosql_apply_parameters_to_query(query, parameters):
...
+    def _render_query(query, parameters):
+        if isinstance(query, dict):
+            return {key: _render_query(value, parameters)
+                    for key, value in deepcopy(query).items()}
+        elif isinstance(query, list):
+            return [_render_query(elt, parameters) for elt in deepcopy(query)]
+        elif type(query) is str:
+            clean_p = deepcopy(parameters)
+            # Add quotes to string parameters to keep type if not complex
+            if (re.compile(r"^{{[^({{)}]*}}$").match(query)
```
Comment (piotch) [inline toucan_connectors/common.py]: maybe compile the regexes in a CONSTANT to make it more readable and probably gain a bit of performance

File: tests/test_common.py [hunk 2]
```diff
@@ -13,53 +13,114 @@ def test_apply_parameter_to_query_do_nothing():
...
+        ]}},
+         {'my_indic': {'zero': '0', 'one': 1, 'two': '2'}},
+         {'$match': {'$and': [
+             {'domain': 'truc', 'indic0': '0'},
+             {'indic1': 1, 'indic2': 'yo_2'},
+             {'indic_list': {'zero': '0', 'one': 1, 'two': '2'}}
+         ]}})
+    ]
+    for test in tests:
+        assert nosql_apply_parameters_to_query(test[0], test[1]) == test[2]
```
Comment (piotch) [inline tests/test_common.py]: maybe name the variables rather that using the index in the tuple to make it more readable

File: tests/test_common.py [hunk 3]
```diff
@@ -13,53 +13,114 @@ def test_apply_parameter_to_query_do_nothing():
...
+def test_apply_parameter_to_query():
     """
-    It should work when a paramters is an int
+    It sould work render all parameters
     """
-    query = [{'$match': {'domain': '%(param1)s', 'cat': '%(param2)s'}}]
-    parameters = {'param1': 'yo', 'param2': 1}
-    expected = [{'$match': {'domain': 'yo', 'cat': 1}}]
-    assert nosql_apply_parameters_to_query(query, parameters) == expected
+    tests = [
```
Comment (piotch) [inline tests/test_common.py]: maybe a comment here showing what is what in all those tuples : 

```
(
  query,
  params,
  expected
)
```

File: tests/test_common.py [hunk 4]
```diff
@@ -63,7 +63,10 @@ def test_apply_parameter_to_query():
          ]}})
     ]
     for test in tests:
```
Comment (adimascio) [inline tests/test_common.py]: what about `for query, params expected in tests`?

**Comment by virobidou** [comment]
[ETQUser, mes permissions sont appliquées à mon requester report](https://trello.com/c/gQXKMxO9/3246-etquser-mes-permissions-sont-appliqu%C3%A9es-%C3%A0-mon-requester-report)

**Pull request merged**